src/plots/cartesian/dragbox.js

```diff
@@ -543,8 +548,12 @@ module.exports = function dragBox(gd, plotinfo, x, y, w, h, ns, ew) {
             pushActiveAxIds(yaLinked);
         }
 
+        updates = {};
         for(i = 0; i < activeAxIds.length; i++) {
-            doTicks(gd, activeAxIds[i], true);
+            var axId = activeAxIds[i];
+            doTicks(gd, axId, true);
+            var ax = getFromId(gd, axId);
+            updates[ax._name + '.range'] = ax.range.slice();
```

Comment:
note that one effect of doing it this way is when you drag axis ends or make a zoombox (any of the GUI interactions) while using `constrain: 'domain'`, the shape and constrained domains of the subplot will never change because we always make proportional updates to all linked ranges. This is NOT the case for generic `relayout` calls: if you `relayout` a single range, it can reshape the subplot, potentially changing which axis is constrained, but always keeping the domains as large as possible within the constraint.